File: internal/plan/plan.go
package plan

import "time"

// Plan represents a collection of tasks extracted from a source document.
type Plan struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	SourceFile  string    `json:"sourceFile"`
	CreatedAt   time.Time `json:"createdAt"`
	Status      string    `json:"status"`
	Tasks       []Task    `json:"tasks"`
}

// Plan status constants
const (
	PlanStatusNotStarted = "not_started"
	PlanStatusInProgress = "in_progress"
	PlanStatusCompleted  = "completed"
	PlanStatusFailed     = "failed"
)

File: ts-port/src/core/plan.ts
/**
 * Plan types and operations
 */

import type { Task } from "./task.js";

export type PlanStatus = "not_started" | "in_progress" | "completed" | "failed";

export interface Plan {
  id: string;
  name: string;
  description: string;
  sourceFile: string;
  createdAt: string;
  status: PlanStatus;
  tasks: Task[];
}

export function createPlan(
  id: string,
  name: string,
  description: string,
  sourceFile: string,
  tasks: Task[]
): Plan {
  return {
    id,
    name,
    description,
    sourceFile,
    createdAt: new Date().toISOString(),
    status: "not_started",
    tasks,
  };
}
